src/util/replace_symbol.cpp

```diff
@@ -25,7 +25,11 @@ void replace_symbolt::insert(
   const symbol_exprt &old_expr,
   const exprt &new_expr)
 {
-  PRECONDITION(old_expr.type() == new_expr.type());
+  PRECONDITION_WITH_DIAGNOSTICS(
+    old_expr.type() == new_expr.type(),
+    "types to be replaced should match.\nold type: '" +
+      old_expr.type().pretty() + "'\nnew.type: '" + new_expr.type().pretty() +
+      "')");
```

Comment:
```suggestion
 PRECONDITION_WITH_DIAGNOSTICS(
    old_expr.type() == new_expr.type(),
    "types to be replaced should match.\nold type: ", old_expr.type(),
    "new.type: ", new_expr.type());
```

I’d recommend against the quotes and what not (the output isn’t really machine parseable anyway, and irep outputs are almost always going to be multiline so the quotes don’t really help with human readability either), and anything that has a specialisation for `diagnostics_helpert` (ireps do), can just be added as an argument and the macro takes care of the concatenation.